MachineLearningPotential/Cutoff.jl

```diff
@@ -0,0 +1,68 @@
+module Cutoff
+
+using LinearAlgebra
+using StaticArrays
+
+export distance2,find_distance2_mat,angular_measure,all_angular_measure
+export cutoff_function
+
+#----------------------------------------------------------------#
+#-------------------------Measurements---------------------------#
+#----------------------------------------------------------------#
+
+#Beginning with basic distance funtions required throughout MLP calculations
+"""
+    distance2(a,b)
+squared distance of two vectors `a` `b` 
+"""
+dist2(a,b) = (a-b)⋅(a-b)
+"""
+    find_distance2_mat_distance2_mat(pos)
+given a vector called `pos` comprised of (ideally) static vectors we return a lengthXlength symmetric matrix of the squared distance
+"""
+find_distance2_mat(pos) = [dist2(a,b) for a in pos, b in pos]
+"""
+    angular_measure(a,b,c,r2ij,r2ik)
+    angular_measure(a,b,c)
+angular measure accepts three vectors `a`,`b`,`c` and can either accept or calculate the squared distances between them `r2_ab`,`r2_bc`, centred on vector `a`. Returns cos(θ) labelled as θ: the angular measure.
+"""
+function angular_measure(a,b,c,r2ab,r2ac)
+    θ = (a - b)⋅(a - c)/sqrt(r2ab*r2ac) 
+    return θ
+end
+function angular_measure(a,b,c)
+    r2_ab,r2_ac,r2_bc = dist2(a,b),dist2(a,c),dist2(b,c)
+    θ = angular_measure(a,b,c,r2_ab,r2_ac)   
+    return θ,r2_ab,r2_ac,r2_bc
+end
+
+thetacalc(xy,xz,disxy,disxz) = xy⋅xz/(disxy*disxz)
+function all_angular_measure(a,b,c,r2ab,r2ac,r2bc)
+    ab,ac,bc = (a-b),(a-c),(b-c)
+    dis2ab,dis2ac,dis2bc = sqrt(r2ab),sqrt(r2ac),sqrt(r2bc)
+    thetavec = [thetacalc(ab,ac,dis2ab,dis2ac),thetacalc(-ab,bc,dis2ab,dis2bc),thetacalc(-ac,-bc,dis2ac,dis2bc) ] 
+
+    return thetavec
+end
+#------------------------------------------------------------------#
+#----------------------Type 2 cutoff function----------------------#
+#------------------------------------------------------------------#
+"""
+    cutoff_function(r_scaled)
+    (r_ij,r_cut)
+    (dist_vec::T,r_cut) where {T<:Array}
```

Comment:
you want to repeat the whole function call, so eg.  cutoff_function(r_ij,r_cut)
and remove the where statement; you can explain the types of the arguments in the next. part of the documentation